core/trino-main/src/test/java/io/trino/security/TestAccessControlManager.java

```diff
@@ -186,6 +190,29 @@ public void testDenyCatalogAccessControl()
         }
     }
 
+    @Test
+    public void testDenyTableFunctionCatalogAccessControl()
+    {
+        try (LocalQueryRunner queryRunner = LocalQueryRunner.create(TEST_SESSION)) {
+            TransactionManager transactionManager = queryRunner.getTransactionManager();
+            AccessControlManager accessControlManager = createAccessControlManager(transactionManager);
+
+            TestSystemAccessControlFactory accessControlFactory = new TestSystemAccessControlFactory("test");
+            accessControlManager.addSystemAccessControlFactory(accessControlFactory);
+            accessControlManager.loadSystemAccessControl("allow-all", ImmutableMap.of());
+
+            queryRunner.createCatalog(TEST_CATALOG_NAME, MockConnectorFactory.create(), ImmutableMap.of());
+            accessControlManager.setConnectorAccessControlProvider(CatalogServiceProvider.singleton(TEST_CATALOG_HANDLE, Optional.of(new DenyConnectorAccessControl())));
```

Comment:
could you please verify that you have access before setting DenyConnectorAccessControl?